k8s/reconcilers/deployment.go

```diff
@@ -0,0 +1,689 @@
+		return false
+	}
+	if summary.Summary.IsRemoval {
+		return object.GetDeletionTimestamp() != nil
+	}
+	return object.GetDeletionTimestamp() == nil
+}
+
+func (r *DeploymentReconciler) deploymentHashMatch(ctx context.Context, object Reconcilable, summary *model.SummaryResult) bool {
+	if object == nil || summary == nil { // we don't expect any of these to be nil
+		return false
+	}
+	deployment, err := r.deploymentBuilder(ctx, object)
+	if err != nil {
+		return false
+	}
+	return summary.DeploymentHash == deployment.Hash
+}
+
+func (r *DeploymentReconciler) queueDeploymentJob(ctx context.Context, object Reconcilable, isRemoval bool, updateCorrelationId bool, operationStartTimeKey string) error {
+	// If previous status was terminal and there is no parity between the summary and current object, then update correlation id.
+	// This will ensure that there is a new correlation id between deployments including deployments that periodically occur.
+	if updateCorrelationId && utilsmodel.IsTerminalState(object.GetStatus().ProvisioningStatus.Status) {
+		r.updateCorrelationIdMetaData(ctx, object, operationStartTimeKey)
+	}
+
+	// Build the deployment object to send to the api
+	deployment, err := r.deploymentBuilder(ctx, object)
+	if err != nil {
+		return err
+	}
+
+	// Send the deployment object to the api to queue a job
+	err = r.apiClient.QueueDeploymentJob(ctx, object.GetNamespace(), isRemoval, *deployment)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *DeploymentReconciler) getDeploymentSummary(ctx context.Context, object Reconcilable) (*model.SummaryResult, error) {
+	return r.apiClient.GetSummary(ctx, r.deploymentKeyResolver(object), object.GetNamespace())
+}
+
+func (r *DeploymentReconciler) updateCorrelationIdMetaData(ctx context.Context, object Reconcilable, operationStartTimeKey string) error {
+	correlationId := uuid.New()
+	r.patchOperationStartTime(object, operationStartTimeKey)
+	object.GetAnnotations()[constants.AzureCorrelationId] = correlationId.String()
+	if err := r.kubeClient.Update(ctx, object); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *DeploymentReconciler) patchOperationStartTime(object Reconcilable, operationStartTimeKey string) {
+	annotations := object.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	annotations[operationStartTimeKey] = time.Now().Format(time.RFC3339)
+	object.SetAnnotations(annotations)
+}
+
+func (r *DeploymentReconciler) ensureOperationState(annotations map[string]string, objectStatus *k8smodel.DeployableStatus, provisioningState string) {
+	objectStatus.ProvisioningStatus.Status = provisioningState
+	objectStatus.ProvisioningStatus.OperationID = annotations[constants.AzureOperationIdKey]
+}
+
+func (r *DeploymentReconciler) updateObjectStatus(ctx context.Context, object Reconcilable, summaryResult *model.SummaryResult, opts patchStatusOptions, log logr.Logger) (provisioningState string, err error) {
```

Comment:
Looks like the first return value provisioningState is not used at all callers. Do you want to remove it? 
What's the expected meaning for this return value - current status or updated status?

Reply:
I think this is intended. The status is the return value of provisioning status even if no callers need to use it. Usually we only need the provisoning status updated in object, but it is fine we still return this value.